src/VisualStudio/Core/Def/Implementation/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -0,0 +1,213 @@
+            var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(reference.SourceSpan, context.CancellationToken).ConfigureAwait(false);
+            var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+            var referenceLocation = await ProtocolConversions.DocumentSpanToLocationAsync(reference.SourceSpan, cancellationToken).ConfigureAwait(false);
+            var docText = await reference.SourceSpan.Document.GetTextAsync(context.CancellationToken).ConfigureAwait(false);
+            var classifiedText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+            var locationWithText = new LSP.LocationWithText { Range = referenceLocation.Range, Uri = referenceLocation.Uri, Text = classifiedText };
+
+            result.ClassifiedContext = 
+            */
+            var mappedDocumentSpan = await TryMapAndGetFirstAsync(documentSpan, sourceText, token).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+                result.PersistentSpan = symbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
+            }
+
+            return result;
+        }
+
+        internal static async Task<RoslynSymbolResult> MakeAsync(RoslynSymbolSource symbolSource, SourceReferenceItem reference, DocumentSpan documentSpan, CancellationToken token)
+        {
+            // TODO: Understand how Roslyn calls GetReferenceGroupsAsync
+            // it seems that at this time, we have full knowledge of definitions
+            // and their references
+
+            var result = await MakeAsync(symbolSource, reference.Definition, documentSpan, token)
+                .ConfigureAwait(false);
+            //result.Name = reference.ToString();
+            return result;
+        }
+
+        public IPersistentSpan PersistentSpan { get; private set; }
+
+        public ClassifiedTextElement ClassifiedDefinition { get; private set; }
+
+        public Span HighlightSpan { get; private set; }
+
+        public string DocumentId { get; private set; }
+
+        public string ProjectId { get; private set; }
+
+        public ClassifiedTextElement ClassifiedContext { get; private set; }
+
+        public ImageId DefinitionIcon { get; private set; }
+
+        public string ProjectName { get; set; }
+
+        public Task NavigateToAsync(CancellationToken token)
+        {
+            // how to get the workspace?
+            //return definition.TryNavigateTo()
```

Comment:
?

Reply:
it will be resolved prior to checking in